### PR TITLE
[Frontend] Feature/auth

### PR DESCRIPTION
**Pull request by kuzkovandrey**
Created base auth feature
- used mock user: email - qwe@qwe.qwe, password: - qweqwe.
- created layouts without mobile adaptive 
- refactored code architecture

### CODE REVIEW

**Comment by dkotlyar** [comment]
Can we move [input-field](https://github.com/kuzkovandrey/podcasts-api/tree/feature/auth/webapp/frontend/src/app/features/auth/components/input-field) and [logo](https://github.com/volunteer-prb/podcasts-api/tree/58c83582b4e74e1dc279ffabac759f23769ffa6a/webapp/frontend/src/app/features/auth/components/logo) to [components](https://github.com/kuzkovandrey/podcasts-api/tree/feature/auth/webapp/frontend/src/app/shared/components) for future reuse?

**Comment by dkotlyar** [comment]
Please use `em` and `rem` instead `px` in styles.

**Comment by dkotlyar** [comment]
You're have duplicate styles between [sign-up.component.scss](https://github.com/volunteer-prb/podcasts-api/blob/58c83582b4e74e1dc279ffabac759f23769ffa6a/webapp/frontend/src/app/features/auth/pages/sign-up/sign-up.component.scss) and [sign-in.component.scss](https://github.com/volunteer-prb/podcasts-api/blob/58c83582b4e74e1dc279ffabac759f23769ffa6a/webapp/frontend/src/app/features/auth/pages/sign-in/sign-in.component.scss), create single meta-component or move style to common file.

File: webapp/frontend/src/app/shared/components/checkbox/checkbox.component.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'checkbox',
+  templateUrl: './checkbox.component.html',
+  styleUrls: ['./checkbox.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CheckboxComponent {
+  @Input() text: string = 'Запомнить пароль';
```
Comment (dkotlyar) [inline webapp/frontend/src/app/shared/components/checkbox/checkbox.component.ts:10]: Remove default text from shared component

Comment (kuzkovandrey) [reply webapp/frontend/src/app/shared/components/checkbox/checkbox.component.ts]: resolved

**Comment by kuzkovandrey** [comment]
> Can we move [input-field](https://github.com/kuzkovandrey/podcasts-api/tree/feature/auth/webapp/frontend/src/app/features/auth/components/input-field) and [logo](https://github.com/volunteer-prb/podcasts-api/tree/58c83582b4e74e1dc279ffabac759f23769ffa6a/webapp/frontend/src/app/features/auth/components/logo) to [components](https://github.com/kuzkovandrey/podcasts-api/tree/feature/auth/webapp/frontend/src/app/shared/components) for future reuse?

At the moment the logo component is only used in the auth feature, as per the design of. If it will be used in another feature, then need to move it to the common directory
Input field for auth feature includes difficult logic with input state, for channels feature i think may be to create simple input field

**Comment by kuzkovandrey** [comment]
> Please use `em` and `rem` instead `px` in styles.

resolved

**Comment by kuzkovandrey** [comment]
> You're have duplicate styles between [sign-up.component.scss](https://github.com/volunteer-prb/podcasts-api/blob/58c83582b4e74e1dc279ffabac759f23769ffa6a/webapp/frontend/src/app/features/auth/pages/sign-up/sign-up.component.scss) and [sign-in.component.scss](https://github.com/volunteer-prb/podcasts-api/blob/58c83582b4e74e1dc279ffabac759f23769ffa6a/webapp/frontend/src/app/features/auth/pages/sign-in/sign-in.component.scss), create single meta-component or move style to common file.

resolved

**Comment by andreinechaev** [comment]
@dkotlyar any other concerns or we're good to go?

**Comment by dkotlyar** [comment]
I don't have any remarks

**Pull request merged**